api/engineimage.go

```diff
@@ -13,7 +13,7 @@ import (
 func (s *Server) EngineImageList(rw http.ResponseWriter, req *http.Request) (err error) {
 	apiContext := api.GetApiContext(req)
 
-	eis, err := s.m.ListEngineImagesByName()
+	eis, err := s.m.ListEngineImagesSorted()
```

Comment:
In fact, why do the resources need to be sorted? Doesn't get it from reading the code.

Reply:
Um... why this comment doesn't show in the PR comment...

Reply:
And we cannot use map since in Rancher API, a collection is an array. Got it, thanks.